File: frontend/src/types/adminRule.ts
export type AdminRuleCategory =
  | "CHEMICAL"
  | "PSM"
  | "DANGER"
  | "ENV"
  | "HEALTH"
  | "FIRE"
  | "ETC";

export interface AdminRuleListItem {
  admrul_sn: string;
  admrul_id: number;
  admrul_name: string;
  admrul_type_name?: string | null;
  ministry_names?: string | null;
  category: AdminRuleCategory;
  issue_date?: string | null;
  enforce_date?: string | null;
  nlic_registered_date?: string | null;
  issue_number?: string | null;
  detail_link_path?: string | null;
}

export interface AdminRuleListResponse {
  total: number;
  items: AdminRuleListItem[];
}

File: backend/app/schemas/admin_rule.py
# app/schemas/admin_rule.py
from datetime import date
from typing import List, Optional

from pydantic import BaseModel
from app.enums.admin_rule_category import AdminRuleCategory


class AdminRuleListItem(BaseModel):
    admrul_sn: str
    admrul_id: int
    admrul_name: str
    admrul_type_name: Optional[str] = None
    ministry_names: Optional[str] = None

    category: AdminRuleCategory

    issue_date: Optional[date] = None
    enforce_date: Optional[date] = None
    nlic_registered_date: Optional[date] = None

    issue_number: Optional[str] = None

    detail_link_path: Optional[str]


class AdminRuleListResponse(BaseModel):
    total: int
    items: List[AdminRuleListItem]
